src/Symfony/Component/Finder/Glob.php

```diff
@@ -60,9 +60,15 @@ public static function toRegex($glob, $strictLeadingDot = true, $strictWildcardS
 
             $firstByte = '/' === $car;
 
-            if ($firstByte && $strictWildcardSlash && isset($glob[$i + 3]) && '**/' === $glob[$i + 1].$glob[$i + 2].$glob[$i + 3]) {
-                $car = $strictLeadingDot ? '/(?:(?=[^\.])[^/]++/)*' : '/(?:[^/]++/)*';
-                $i += 3;
+            if ($firstByte && $strictWildcardSlash) {
+                if (isset($glob[$i + 3]) && '**/' === $glob[$i + 1].$glob[$i + 2].$glob[$i + 3]) {
+                    $car = $strictLeadingDot ? '/(?:(?=[^\.])[^/]++/)*' : '/(?:[^/]++/)*';
+                    $i += 3;
+                } elseif (isset($glob[$i + 2]) && '**' === $glob[$i + 1].$glob[$i + 2]) {
+                    $car = $strictLeadingDot ? '/(?=[^\.]).*' : '/.*';
```

Comment:
to be, the rx here should be the same as before, meaning the patch could have just onle line:
`if ($firstByte && $strictWildcardSlash && isset($glob[$i + 2]) && '**' === $glob[$i + 1].$glob[$i + 2] && (!isset($glob[$i + 3]) || '/' === $glob[$i + 3])) {`